sdk/search/search-documents/src/serviceModels.ts

```diff
@@ -852,7 +852,7 @@ export interface SynonymMap {
    * keys is not available for free search services, and is only available for paid services
    * created on or after January 1, 2019.
    */
-  encryptionKey?: SearchResourceEncryptionKey;
+  encryptionKey?: SearchResourceEncryptionKey | null;
```

Comment:
I don't really understand what `null` means for any of these

Reply:
Provided a detailed explanation at the end.